pallets/bridge_transfer/src/lib.rs

```diff
@@ -123,11 +110,62 @@ decl_module! {
 		pub fn lottery_handler(origin, metadata: Vec<u8>, rid: ResourceId) -> DispatchResult {
 			T::BridgeOrigin::ensure_origin(origin)?;
 
-			Self::deposit_event(Event::LotteryCommand(metadata));
-
-			// TODO: Send to BTC Lottery pallet to handle this
+			let op = u8::from_be_bytes(<[u8; 1]>::try_from(&metadata[..1]).map_err(|_| Error::<T>::InvalidCommand)?);
+			if op == 0 {
+				ensure!(
+					metadata.len() == 13,
+					Error::<T>::InvalidCommand
+				);
+
+				Self::deposit_event(Event::LotteryNewround(
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[1..5]).map_err(|_| Error::<T>::InvalidCommand)?),	// roundId
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[5..9]).map_err(|_| Error::<T>::InvalidCommand)?),	// totalCount
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[9..]).map_err(|_| Error::<T>::InvalidCommand)?)	// winnerCount
+				));
+			} else if op == 1 {
+				ensure!(
+					metadata.len() > 13,
+					Error::<T>::InvalidCommand
+				);
+
+				let address_len: usize = u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[9..13]).map_err(|_| Error::<T>::InvalidCommand)?).saturated_into();
+				ensure!(
+					metadata.len() == (13 + address_len),
+					Error::<T>::InvalidCommand
+				);
+
+				Self::deposit_event(Event::LotteryOpenbox(
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[1..5]).map_err(|_| Error::<T>::InvalidCommand)?),	// roundId
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[5..9]).map_err(|_| Error::<T>::InvalidCommand)?),	// tokenId
+					metadata[13..].to_vec()						// btcAddress
+				));
+			}
 
 			Ok(())
 		}
 	}
 }
+
+impl<T: Config> Module<T> {
+	pub fn transfer_lottery(round_id: u32, token_id: u32, payload: Vec<u8>, dest_id: bridge::ChainId) -> DispatchResult {
```

Comment:
Consider rename to `do_transfer_lottery`, follow the pattern https://github.com/paritytech/polkadot/blob/master/runtime/common/src/paras_registrar.rs#L219

Reply:
Reamed